homeassistant/components/owlet/__init__.py

```diff
@@ -0,0 +1,83 @@
+"""
+Support for Owlet baby monitors.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/owlet/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (CONF_USERNAME, CONF_PASSWORD, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+
+
+REQUIREMENTS = ['pyowlet==1.0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'owlet'
+
+SENSOR_TYPES = [
+    'oxygen_level',
+    'heart_rate',
+    'base_station_on',
+    'movement'
+]
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_NAME): cv.string,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up owlet component."""
+    from pyowlet.PyOwlet import PyOwlet
+
+    username = config[DOMAIN].get(CONF_USERNAME)
+    password = config[DOMAIN].get(CONF_PASSWORD)
+    name = config[DOMAIN].get(CONF_NAME)
+
+    device = PyOwlet(username, password)
+
+    device.update_properties()
+
+    if not name:
+        name = '{}\'s Owlet'.format(device.baby_name)
+
+    hass.data[DOMAIN] = OwletDevice(device, name, SENSOR_TYPES)
+
+    load_platform(hass, 'sensor', DOMAIN, None, config)
+    load_platform(hass, 'binary_sensor', DOMAIN, None, config)
+
+    return True
+
+
+class OwletDevice():
+    """Represents a configured Owlet device."""
+
+    def __init__(self, device, name, monitor):
+        """Initialize device."""
+        self._name = name
+        self._monitor = monitor
+        self._device = device
+
+    @property
+    def name(self):
```

Comment:
We don't need properties for these attributes. Just use regular instance attributes.